lua/jdtls/junit.lua

```diff
@@ -78,43 +76,111 @@ local function mk_buf_loop(sock, handle_buffer)
   end
 end
 
-
 function M.mk_test_results(bufnr)
+  vim.api.nvim_buf_clear_namespace(bufnr, ns, 0, -1)
+  vim.diagnostic.reset(ns, bufnr)
   local tests = {}
+
   local handle_buffer = function(buf)
     parse(buf, tests)
   end
+
+  local function get_test_start_line_num(lenses, test)
+    if test.method ~= nil then
+      if #lenses > 0 then
+        for _, v in ipairs(lenses) do
+          if vim.startswith(v.label, test.method) then
+            return v.range.start.line
+          end
+        end
+      else
+        if vim.startswith(lenses.label, test.method) then
+          return lenses.range.start.line
+        end
+      end
+    end
+    return nil
+  end
+
   return {
-    show = function()
+    show = function(lens)
       local items = {}
       local repl = require('dap.repl')
       local num_failures = 0
+      local lenses = lens.children or lens
+      local failures = {}
+      local error_symbol = '❌'
+      local success_symbol = '✔️ '
```

Comment:
It may be possible, but, I haven't looked much into configuration aspect. Do you have different symbols in mind that provide better context as default?